mongo/test_mongo.py

```diff
@@ -157,6 +156,40 @@ def test_state_translation(self):
         unknown_desc = self.check.get_state_description(500)
         self.assertTrue(unknown_desc.find('500') != -1)
 
+    def test_server_uri_sanitization(self):
+        # Initialize check
+        config = {
+            'instances': [self.MONGODB_CONFIG]
+        }
+        self.load_check(config)
+
+        _parse_uri = self.check._parse_uri
+
+        # Batch with `sanitize_username` set to False
+        server_names = (
+            ("mongodb://localhost:27017/admin", "mongodb://localhost:27017/admin"),
+            ("mongodb://user:pass@localhost:27017/admin", "mongodb://user:*****@localhost:27017/admin"),
+            ("mongodb://user:pass_%2@localhost:27017/admin", "mongodb://user:*****@localhost:27017/admin"), # pymongo parses the password as `pass_%2`
+            ("mongodb://user:pass_%25@localhost:27017/admin", "mongodb://user:*****@localhost:27017/admin"), # pymongo parses the password as `pass_%` (`%25` is url-decoded to `%`)
+        )
+
+        for server, expected_clean_name in server_names:
+            _, _, _, _, clean_name = _parse_uri(server, sanitize_username=False)
+            self.assertEquals(expected_clean_name, clean_name)
+
+
+        # Batch with `sanitize_username` set to True, with this option no password is expected in the mongo URI
+        server_names = (
+            ("mongodb://localhost:27017/admin", "mongodb://localhost:27017/admin"),
```

Comment:
I understand that sanitize_username should also sanitize `username:password`... I understand that use-case would be viable(?), at least as far as `_parse_uri()` is concerned. We're missing that test case case here:

```
("mongodb://user%25:mybadasspass@localhost:27017/admin", "mongodb://localhost:27017/admin"),
```

I'm not 100% sure but the current code might actually not match in that case [here](https://github.com/DataDog/integrations-core/pull/347/files#diff-6e728262c9cc478d4678e74087b5e01dR611)